drkafka/src/main/java/com/pinterest/doctorkafka/modules/action/cluster/ScriptReplaceInstance.java

```diff
@@ -0,0 +1,147 @@
+package com.pinterest.doctorkafka.modules.action.cluster;
+
+import com.pinterest.doctorkafka.modules.action.Action;
+import com.pinterest.doctorkafka.modules.errors.ModuleConfigurationException;
+import com.pinterest.doctorkafka.modules.event.Event;
+import com.pinterest.doctorkafka.modules.event.NotificationEvent;
+import com.pinterest.doctorkafka.modules.event.ReportEvent;
+import com.pinterest.doctorkafka.util.ZookeeperClient;
+
+import org.apache.commons.configuration2.AbstractConfiguration;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class ScriptReplaceInstance extends Action implements Runnable {
+  private static final Logger LOG = LogManager.getLogger(ScriptReplaceInstance.class);
+
+  private static final String CONFIG_SCRIPT_KEY = "script";
+  private static final String CONFIG_PROLONG_REPLACEMENT_ALERT_SECONDS_KEY = "prolong.replacement.alert.seconds";
+
+  private String configScript;
+  private long configProlongReplacementAlertSeconds = 1800L;
+
+  private static final String EVENT_NOTIFY_PROLONG_REPLACEMENT_NAME = "notify_prolong_replacement";
+  private static final String EVENT_NOTIFY_REPLACEMENT_NAME = "notify_replacement";
+  private static final String EVENT_REPORT_OPERATION_NAME = "report_operation";
+
+  private static final String EVENT_CLUSTER_NAME_KEY = "cluster_name";
+  private static final String EVENT_HOSTNAME_KEY = "hostname";
+  private static final String EVENT_ZOOKEEPER_CLIENT_KEY = "zookeeper_client";
+
+  private static final String DEFAULT_EVENT_SUBJECT = "n/a";
+
+  private volatile boolean isBusy = false;
+  private String currentHostname;
+  private long replacementStartTime = -1;
+  private Thread thread;
+
+  @Override
+  public void configure(AbstractConfiguration config) throws ModuleConfigurationException {
+    super.configure(config);
+    if (!config.containsKey(CONFIG_SCRIPT_KEY)){
+      throw new ModuleConfigurationException("Missing config " + CONFIG_SCRIPT_KEY + " for plugin " + this.getClass());
+    }
+    this.configScript = config.getString(CONFIG_SCRIPT_KEY).replaceAll("^\"|\"$","");
+    this.configProlongReplacementAlertSeconds = config.getLong(
+        CONFIG_PROLONG_REPLACEMENT_ALERT_SECONDS_KEY,
+        configProlongReplacementAlertSeconds
+    );
+  }
+
+  @Override
+  public Collection<Event> execute(Event event) throws Exception {
+    if(isDryRun()){
+      LOG.info("Dry run: Action {} triggered by event {}", this.getClass(), event.getName());
```

Comment:
Could you refactor the isDryRun() check? Doesn't make sense to check inside every action.